.github/workflows/build.yml

```diff
@@ -67,6 +67,25 @@ jobs:
       working-directory: build/check_install_root
       run: diff -u ../../test/jni_list.txt <(du -a | tac | awk -F ' ' '{print $2}' | sort)
 
+
+  build-on-ubuntu-for-python:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: asdf-vm/actions/install@v1
+    - name: Install dependencies
+      run: |
+        python3 -m pip install --upgrade pip
+        if [ -f requirements.txt ]; then pip3 install -r requirements.txt; fi
+    - name: Configure cmake
+      run: cmake -S . -B build -DDJINNI_WITH_PYTHON=ON
+    - name: Build release
+      run: cmake --build build --parallel $(nproc) --config Release
+    - name: Run tests
+      working-directory: build/test-suite
+      run: ctest -C Release -V
```

Comment:
Not that it's necessary but I think it could be possible/useful to combine the macos und ubuntu python builds by adding another dimension to the build matrix. 🤔  Like this:

```yaml
build-for-python:
    strategy:
      matrix:
        python-version: [3.6, 3.9]
        os: [macos-10.15, ubuntu-latest]
    runs-on: ${{ matrix.os }}
    steps:
...
    - name: Set up Python ${{ matrix.python-version }}
      uses: actions/setup-python@v2
      with:
        python-version: ${{ matrix.python-version }}
```

Reply:
yes, but `$(nproc)` is platform specific, (not without some extra action on mac) 
but if this could be created into a variable, then we could have one matrix

Reply:
We actually know the number of CPU cores is [always 2 for Ubuntu & 3 for macOS](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources) on github actions. Maybe it's easier to just hardcode it to `--parallel 2`? 
But as I said this is not critical to me, we can also do this in a later PR, just want to discuss this to explore what's possible.

Reply:
I am in general against hardcoding such things

This is some old debug code from a Azure pipleine , maybe we can utilize that in some way and export that into the environment, this might be general useful 

```
jobs:
- job: Linux
  pool:
    vmImage: 'ubuntu-latest'    
  steps:
  - script: |
      ncpus=$(nproc)
      echo "##vso[task.setvariable variable=number_cpus]$ncpus"
    displayName: 'explore environment'
  - script: |
      echo "HERE WITH $(number_cpus)"
    displayName: 'number of cores'

- job: macOS
  pool:
    vmImage: 'macOS-latest'
  steps:  
  - script: |
      ncpus=$(sysctl -n hw.ncpu)
      echo "##vso[task.setvariable variable=number_cpus]$ncpus"
    displayName: 'explore environment'  
  - script: |
      echo "HERE WITH $(number_cpus)"
    displayName: 'number of cores'

- job: Windows
  pool:
    vmImage: 'windows-latest'
  steps:
  - script: |
      echo "##vso[task.setvariable variable=number_cpus]%NUMBER_OF_PROCESSORS%"
    displayName: 'explore environment'  
  - script: |
      echo "HERE WITH $(number_cpus)"
    displayName: 'number of cores'

```

Reply:
but that might become an extra issue